e-commerce/api/server.ts

```diff
@@ -0,0 +1,65 @@
+import { Stripe } from 'stripe'
+import { auth } from '@tensei/auth'
+import { graphql } from '@tensei/graphql'
+import { welcome, tensei, cors, resource, text, textarea, integer, graphQlQuery } from '@tensei/core'
+
+const stripe = new Stripe('sk_test_51JjsxtJnZnb1n2GnbebKQgYe73xTZtT1147z1dLFU9CGmkfbZKvcrW1VX4jA10Wav82slFIGjp9AW6puPURjibMP00z7Fw14Em', {
+  apiVersion: '2020-08-27'
+})
+
+interface ProductCartItem {
+  productId: string
+  quantity: number
+}
+
+export default tensei()
+  .root(__dirname)
+  .graphQlTypeDefs([`
+    input ProductCartItem {
+      productId: ID
+      quantity: Int
+    }
+
+    extend type Mutation {
+      createPaymentIntent(productCartItems: [ProductCartItem]!): String!
+    }
+  `])
+  .graphQlQueries([
+    graphQlQuery('Create payment intent')
+      .path('createPaymentIntent')
+      .mutation()
+      .handle(async (src, args, ctx, info) => {
+        const products = await ctx.repositories.products().find({
+          id: {
+            $in: ctx.body.productCartItems.map((item: ProductCartItem) => item.productId)
+          }
+        })
+
+        const totalPrice = products.reduce((total:any, product:any)  => total + parseInt(product.price), 0)
```

Comment:
Let's not use `any` here. total is `number` and product is `ProductModel`. You can import `ProductModel` from '@tensei/orm`.

Reply:
Done